### PR TITLE
Add `JsonBone`

### PR DESCRIPTION
**Pull request by ArneGudermann**
This draft implements a basic Json bone.
Have anyone ideas for it please let me know.

### CODE REVIEW

File: core/bones/json.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        if name not in skel.accessedValues:
+            return False
+
+        value = skel.accessedValues[name]
+        if isinstance(value, str):
+            try:
+                _ = json.loads(value)
+                del _
+            except Exception as e:
+                raise f"Error in serialize in JsonBone: {e=}"
```
Comment (sveneberth) [inline core/bones/json.py:28]: You cannot raise a str. But catching an exception and raising another doesn't make sense to me here.

**Review by sveneberth — CHANGES_REQUESTED** [review]

File: core/bones/json.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+        return False
+
+    def singleValueFromClient(self, value, *args, **kwargs):
+        if value:
+            value = str(value)  # Try to parse a JSON string
+            try:
+                value = json.loads(value)
+            except Exception as e:
+                raise f"Error in singleValueFromClient in JsonBone: {e=}"
```
Comment (sveneberth) [inline core/bones/json.py:52]: Here as well

File: core/bones/json.py (hunk 1 above)
Comment (phorward) [reply core/bones/json.py]: Please raise a ValueError

**Review by sveneberth — APPROVED** [review]
Basically it's okay.

But there are some edge cases: You can save `true` (boolean), but not `false` or `null`.

**Comment by phorward** [comment]
> But there are some edge cases: You can save `true` (boolean), but not `false` or `null`.

yes you're right. This issue must be resolved first.

File: core/bones/json.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    type = "raw.json"
+
+    def __init__(self, indexed=False, multiple=False, languages=None, *args, **kwargs):
+        assert not multiple
+        assert not languages
+        assert not indexed
+        super().__init__(*args, **kwargs)
+
+    def serialize(self, skel, name, parentIndexed):
+        if value := skel.accessedValues.get(name):
```
Comment (phorward) [inline core/bones/json.py:21]: This is too lazy, as @sveneberth mentioned in https://github.com/viur-framework/viur-core/pull/558#pullrequestreview-1242394347.

**Review by phorward — CHANGES_REQUESTED** [review]

File: core/bones/json.py (hunk 3 above)
Comment (ArneGudermann) [reply core/bones/json.py:21]:
```suggestion
        if name in skel.accessedValues:
            skel.dbEntity[name] = json.dumps(skel.accessedValues[name])
```
Maybe so ?

File: core/bones/json.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    def __init__(self, indexed=False, multiple=False, languages=None, *args, **kwargs):
+        assert not multiple
+        assert not languages
+        assert not indexed
+        super().__init__(*args, **kwargs)
+
+    def serialize(self, skel, name, parentIndexed):
+        if name in skel.accessedValues:
+            skel.dbEntity[name] = json.dumps(skel.accessedValues[name])
+            skel.dbEntity[name] = json.dumps(value)
```
Comment (sveneberth) [inline core/bones/json.py:23]: This line is too much.
```suggestion
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

**Comment by sveneberth** [comment]
@ArneGudermann Can you add type hints to the methods?

**Review by sveneberth — APPROVED** [review]

**Review by phorward — APPROVED** [review]

**Review by phorward — APPROVED** [review]

**Pull request merged**